addusers.py

```diff
@@ -201,7 +203,106 @@ def exists_in_keystone(check_thing, keystone_things):
         return None
 
 
-def parse_rows(rows, select_user=None):
+def remove_duplicates(rows, UC, PC, EC):
+    """Deletes all duplicate rows.
+    user_id_column: UC
+    new_project_column: PC
+    existing_project_column: EC
+    """
+    if len(rows) <= 1:
+        return rows
+    else:
+        output = []
+        for i in range(len(rows)):
+            flag = True
+            for j in rows[:i]:
+                if rows[i][UC] == j[UC] and\
+                   rows[i][PC] == j[PC] and\
+                   rows[i][EC] == j[EC]:
+                    flag = False
+            if flag:
+                output.append(rows[i])
+    return output
+
+
+def create_new_users_rows(rows):
+    output = []
+    for entry in rows:
+        if entry[7] == 'No':
+            user = [entry[4], entry[5], entry[6], entry[8],
```

Comment:
There should be a more readable way to do this with the User class, that long list of `entry[x]` will be obnoxious to maintain.  It's also not very efficient to iterate again here - you could do this work at the end of the `create_user()` function.

Same thing goes for `create_new_projects_rows()` below